code/modules/ai/ai_behaviors/human_mobs/self_heal.dm

```diff
@@ -0,0 +1,235 @@
+	/obj/item/reagent_containers/hypospray/autoinjector/rezadone,
+	/obj/item/reagent_containers/hypospray/autoinjector/elite,
+))
+
+GLOBAL_LIST_INIT(ai_pain_heal_items, list(
+	/obj/item/reagent_containers/pill/tramadol,
+	/obj/item/reagent_containers/hypospray/autoinjector/tramadol,
+	/obj/item/reagent_containers/hypospray/advanced/tramadol,
+	/obj/item/reagent_containers/hypospray/autoinjector/combat,
+	/obj/item/reagent_containers/hypospray/advanced/oxycodone,
+	/obj/item/reagent_containers/hypospray/autoinjector/oxycodone,
+	/obj/item/reagent_containers/hypospray/advanced/combat_advanced,
+	/obj/item/reagent_containers/hypospray/autoinjector/combat_advanced,
+	/obj/item/reagent_containers/hypospray/autoinjector/neuraline,
+	/obj/item/reagent_containers/hypospray/autoinjector/russian_red,
+	/obj/item/reagent_containers/hypospray/autoinjector/elite,
+))
+
+GLOBAL_LIST_INIT(ai_damtype_to_heal_list, list(
+	BRUTE = GLOB.ai_brute_heal_items,
+	BURN = GLOB.ai_burn_heal_items,
+	TOX = GLOB.ai_tox_heal_items,
+	OXY = GLOB.ai_oxy_heal_items,
+	CLONE = GLOB.ai_clone_heal_items,
+	PAIN = GLOB.ai_pain_heal_items,
+))
+
+/datum/ai_behavior/human
+	///Chat lines for trying to heal
+	var/list/healing_chat = list("Healing, cover me!", "Healing over here.", "Where's the damn medic?", "Medic!", "Treating wounds.", "It's just a flesh wound.", "Need a little help here!", "Cover me!.")
+	///Chat lines for retreating on low health
+	var/list/retreating_chat = list("Falling back!", "Cover me, I'm hit!", "I'm hit!", "Medic!", "Disengaging!", "Help me!", "Need a little help here!", "Tactical withdrawal.", "Repositioning.")
+
+///Reacts if the mob is below the min health threshold
+/datum/ai_behavior/human/proc/check_for_critical_health(datum/source, damage)
+	SIGNAL_HANDLER
+	if(current_action == MOVING_TO_SAFETY)
+		return
+	var/mob/living/living_mob = mob_parent
+	if(!can_heal || living_mob.health - damage > minimum_health * living_mob.maxHealth)
+		return
+	if(mob_parent.incapacitated() || mob_parent.lying_angle) //todo: maybe remove or change this when we add team healing
+		return
+	var/atom/next_target = get_nearest_target(mob_parent, target_distance, TARGET_HOSTILE, mob_parent.faction, need_los = TRUE)
+	if(!next_target)
+		INVOKE_ASYNC(src, PROC_REF(try_heal))
+		return
+	if(prob(50))
+		try_speak(pick(retreating_chat))
+	target_distance = 15
+	change_action(MOVING_TO_SAFETY, next_target, INFINITY)
+
+///Will try healing if possible
+/datum/ai_behavior/human/proc/try_heal()
+	if(prob(75))
+		try_speak(pick(healing_chat))
+	var/mob/living/living_parent = mob_parent
+
+	if(living_parent.on_fire)
+		//extinguish proc so they use an extinguisher if they got it?
+		living_parent.do_resist()
+		return
+
+	var/list/dam_list = list(
+		BRUTE = living_parent.getBruteLoss(),
+		BURN = living_parent.getFireLoss(),
+		TOX = living_parent.getToxLoss(),
+		OXY = living_parent.getOxyLoss(),
+		CLONE = living_parent.getCloneLoss(),
+		PAIN = 0,
+	)
```

Comment:
lol

Reply:
cost of pain being at carbon(human?) level